common/logSanitizer.go

```diff
@@ -99,5 +100,19 @@ func init() {
 		// Regex has two groups: first gets key and delimiter.
 		// Second group gets as many chars as possible that do not terminate the value.
 		sensitiveRegexMap[key] = regexp.MustCompile("(?i)(?P<key>" + key + "[ \t]*[:=][ \t]*)(?P<value>[^& ,;\t\n\r]+)")
+
+		// see comment below
+		if key == strings.ToLower(SigAzure) {
+			mapContainsAzureSig = true
+		}
+	}
+
+	// Double check.
+	// We can't directly do strings.ToLower(SigAzure) in the declaration of
+	// sensitiveRegexMap itself, since it won't be ready if something else logs before that
+	// initialization has happened (as can be the case e.g. init of lifecylemanager).
```

Comment:
a bit confused by this, what's the bug and how is this fixing it?

Reply:
Ha ha, I was confused too when I saw it. The bug goes like this:

1. lifecycleManager has a global singleton called lcm that is initialized by an in-line func.  var lcm =  func(){…}()
2. lifecycleManager.go comes alphabetically before logSanitizer.go, so its variable initialization happens before any initialization in logSanitizer.go
3. The in-line func that initialize the variable in lifecylceManager triggers a glcm Info message if it is enabling perf profiling.
4. That hits the logSanitizer … _before_ the variable init in logSanitizer has run.
5. So logSanitizer's `sensitiveQueryStringKeys`  is not fully initialized at that point. Specifically, it has its correct size (3 elements), and any elements that are constants have their correct values. BUT any element that requires code to initizalize it (e.g. `strings.ToLower(SigAzure)`) has value "" because the code that initializes it _has not yet run_ (as per 2).
6. So the glcm.Info call, into logSanitizer, fails, when it hits that "" value.

Another possible fix would have been to move the lcm init to an actual 'init' function (since those run after all variable initialization) but that only solves this one instance of the issue, and won't protect against any other recurrences if other in-line variable initialization functions trigger sanitization.  So if figured it was better just to make everything constants.

Reply:
PS. I don't actually recall seeing any documentation that says that the constants will all "be there" in a variable like `sensitiveQueryStrings` even before any `code` for it's initialization has run.  However, that's the behavior I observe, and it does seem to make sense logically (e.g. compiler pops the constants in)

Reply:
Ahhh I see, that's certainly not a trivial bug to figure out. Glad you got it.